zerver/lib/validator.py

```diff
@@ -582,3 +582,15 @@ def check_string_or_int(var_name: str, val: object) -> Union[str, int]:
         return val
 
     raise ValidationError(_("{var_name} is not a string or integer").format(var_name=var_name))
+
+
+def check_or(sub_validator1: Validator[Any], sub_validator2: Validator[Any]) -> Validator[Any]:
+    def f(var_name: str, val: object) -> Validator[Any]:
+        try:
+            return sub_validator1(var_name, val)
+        except ValidationError:
+            pass
+
+        return sub_validator2(var_name, val)
```

Comment:
I think this would be cleaner as an interface to extend as a `check_or(*validators: List[Validator[Any]])`, where you can pass an arbitrary number of validators.  I merged this first commit as cd93d0967f0ddad61414fb8fff7445e4efcbd3a3 since it's useful as written, but FYI.

Also, can we make some of the `check_string_or_int` type functions to be implemented using this?  Might be best to do those follow-ups as a small separate PR, since we've merged what is required for this feature.